File: you2.py
from collections import deque

def read_data(filename):
    with open(filename) as f:
        lines = f.readlines()
        n = int(lines[2].split("=")[1])
        data = [tuple(map(int, line.strip().split())) for line in lines[4:]]
        return n, data

def write_data(components):
    for component in components:
        print(" ".join(str(node + 1) for node in component))

def find_connected_components(n, data):
    graph = {i: set() for i in range(n)}
    for x, y in data:
        graph[x - 1].add(y - 1)
        graph[y - 1].add(x - 1)

    visited = set()
    components = []
    for i in range(n):
        if i not in visited:
            component = set()
            bfs(i, graph, visited, component)
            components.append(sorted(component))

    return components

def bfs(start, graph, visited, component):
    queue = deque()
    visited.add(start)
    queue.append(start)

    while queue:
        node = queue.popleft()
        component.add(node)
        for neighbor in graph[node]:
            if neighbor not in visited:
                visited.add(neighbor)
                queue.append(neighbor)

for arqs in range(120):
    print()
    print(f"Arquivo {arqs}")
    n, data = read_data(f"{arqs}.in")
    components = find_connected_components(n, data)
    write_data(components)
    print()

File: Trab1/you2.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <queue>
#include <set>
#include <algorithm>

using namespace std;

void read_data(const string& filename, int& n, vector<pair<int, int>>& data) {
    ifstream file(filename);
    vector<string> lines;
    string line;
    while (getline(file, line)) {
        lines.push_back(line);
    }
    file.close();

    n = stoi(lines[2].substr(2));
    for (int i = 4; i < lines.size(); i++) {
        int x, y;
        sscanf(lines[i].c_str(), "%d %d", &x, &y);
        data.push_back(make_pair(x - 1, y - 1));
    }
}

void write_data(const vector<vector<int>>& components) {
    for (const auto& component : components) {
        for (int node : component) {
            cout << node + 1 << " ";
        }
        cout << endl;
    }
}

void find_connected_components(int n, const vector<pair<int, int>>& data, vector<vector<int>>& components) {
    vector<set<int>> graph(n);
    for (const auto& edge : data) {
        graph[edge.first].insert(edge.second);
        graph[edge.second].insert(edge.first);
    }

    set<int> visited;
    for (int i = 0; i < n; i++) {
        if (visited.count(i) == 0) {
            set<int> component;
            queue<int> q;
            q.push(i);
            visited.insert(i);

            while (!q.empty()) {
                int node = q.front();
                q.pop();
                component.insert(node);
                for (int neighbor : graph[node]) {
                    if (visited.count(neighbor) == 0) {
                        visited.insert(neighbor);
                        q.push(neighbor);
                    }
                }
            }

            vector<int> component_vec(component.begin(), component.end());
            sort(component_vec.begin(), component_vec.end());
            components.push_back(component_vec);
        }
    }
}

int main() {
    for (int arqs = 0; arqs < 120; arqs++) {
        cout << endl;
        cout << "Arquivo " << arqs << endl;
        int n;
        vector<pair<int, int>> data;
        read_data("exemplos/instancias/" + to_string(arqs) + ".in", n, data);
        vector<vector<int>> components;
        find_connected_components(n, data, components);
        write_data(components);
        cout << endl;
    }
    return 0;
}